client/src/containers/user-profile-forms/component.tsx

```diff
@@ -0,0 +1,107 @@
+import { useCallback, useState } from 'react';
+
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import { useMutation } from 'react-query';
+
+import { apiService } from 'services/api';
+import { Label, Input } from 'components/forms';
+import { Button } from 'components/button';
+import { formProps, UserProfilePayload } from './types';
+
+const schemaValidation = yup.object({
+  fname: yup.string().required('first name is required'),
+  lname: yup.string().required('last name is required'),
+  email: yup.string().email().required(),
+});
+
+const editUserData = (data: UserProfilePayload) => apiService.patch('/users/me', data);
+
+const UserDataForm: React.FC<formProps> = ({ alert, showAlert }: formProps) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schemaValidation),
+  });
+
+  const mutation = useMutation(editUserData, {
+    onSuccess: () => {
+      setIsLoading(false);
+
+      alert({
+        type: 'success',
+        title: 'Your changes were successfully saved.',
+      });
+
+      showAlert(true);
+    },
+
+    onError: () => {
+      setIsLoading(false);
+
+      alert({
+        type: 'error',
+        title: 'An error has occured while saving. Please try gain later',
```

Comment:
The error should come from API message. This error could be only when message is undefined